src/Provider/Apple.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/*!
+* Hybridauth
+* https://hybridauth.github.io | https://github.com/hybridauth/hybridauth
+*  (c) 2017 Hybridauth authors | https://hybridauth.github.io/license.html
+*/
+
+namespace Hybridauth\Provider;
+
+use Hybridauth\Adapter\OAuth2;
+
+/**
+ * Apple OAuth2 provider adapter.
+ *
+ * Example:
+ *
+ *   $config = [
+ *       'callback' => Hybridauth\HttpClient\Util::getCurrentUrl(),
+ *       'keys'     => [ 'private_key' => '', 'id' => '', 'team_id' => '', 'key_id' => '', 'secret' => '' ],
```

Comment:
If you don't want to generate the secret inside the Apple provider class (which would be **much** more comfortable to use, but adds dependencies and might have a performance impact), only `secret` and `id` are required here.